### PR TITLE
Improve systemd service file

### PR DESCRIPTION
**Pull request by Flowdalic**
This removes the redundant 'Group=' directive and replaces the deprecated 'PermissionsStartOnly' directive.

### CODE REVIEW

File: contrib/init/bitcoind.service [hunk 1]
```diff
@@ -36,7 +35,6 @@ TimeoutStopSec=600
 
 # Run as bitcoin:bitcoin
 User=bitcoin
-Group=bitcoin
```
Comment (hebasto) [inline contrib/init/bitcoind.service:39]: This group is useful to access the `bitcoin` user's stuff via group permissions.

File: contrib/init/bitcoind.service [hunk 2]
```diff
@@ -20,8 +20,7 @@ ExecStart=/usr/bin/bitcoind -daemon \
                             -datadir=/var/lib/bitcoind
 
 # Make sure the config directory is readable by the service user
-PermissionsStartOnly=true
-ExecStartPre=/bin/chgrp bitcoin /etc/bitcoin
+ExecStartPre=+/bin/chgrp bitcoin /etc/bitcoin
```
Comment (hebasto) [inline contrib/init/bitcoind.service:23]: Is this syntax available on all of the supported platforms?

File: contrib/init/bitcoind.service (hunk 1 above)
Comment (Flowdalic) [reply contrib/init/bitcoind.service:39]: That reads like you assume I've removed the group, which is not entirely true. Systemd will run the process under the user's default group if no group is explicitly specified. And the default group will most likely be the right group (and it definitely should be).

File: contrib/init/bitcoind.service (hunk 2 above)
Comment (Flowdalic) [reply contrib/init/bitcoind.service:23]: Is there list of supported platforms?

Irregardless if there is a list or not, I assume the answer to be yes. Older Linux distribution versions with older systemd, tend to only package older bitcoin versions. The syntax was deprecated with systemd 241, which is currently in 
Debian stable. Likely the replacement syntax was established multiple releases before that. And even considering the rare (and very unlikely) case that you run this new bitcoin version under a very old systemd version, then the distribution package maintainer can remedy this (or the user itself).

**Comment by hebasto** [comment]
The replacement of deprecated `PermissionsStartOnly` was also suggested in #16994.

**Review by hebasto — CHANGES_REQUESTED** [review]
Agree on _"init: remove Group= as it will default to the user's default group"_ commit.

https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Credentials:
> If no group is set, the default group of the user is used.

---

As `ExecStart=+` syntax is used since [v231](https://github.com/systemd/systemd/pull/3797), and basing on discussion in #16994, I suggest to follow @laanwj's [comment](https://github.com/bitcoin/bitcoin/pull/16994#issuecomment-547836888) until CentOS 7 EOL at least as it [ships](url) `systemd` v219 only.

**Comment by Flowdalic** [comment]
I'd argue that (upstream) software's contributed configuration files should contain best-practices, and not worry about compatibility with arcane third-party software. I'd expect the CentOS/RedHat 7 bitcoind packager to test the package and fix any compatibility issues.

**Comment by laanwj** [comment]
The init files are only there as example / documentation of how to use bitcoind as a service, there is no strict requirement for compatibility with everything under the sun, but also no reason to break compatibility for no good reason. It's there as a convenience to people, not as a display of systemd best practices.

That said I'm neutral on this.

**Comment by Flowdalic** [comment]
Rebased on the latest master, please consider for inclusion.

**Comment by achow101** [comment]
Closing this as it has not had any activity in a while. If you are interested in continuing work on this, please leave a comment so that it can be reopened.

**Comment by fanquake** [comment]
PIcked up in #33044.

**Pull request closed without merging**